### DISCUSSION TITLE
A first look to version 1.2...

### DISCUSSION THREAD

**Discussion by Macronair** [Maerciez/TheMillionaireGame#76] [General]
Hi all,

First of all, i'm sorry to keep you all waiting (again! ;D) for a new release or update. Because this is not my only project i'm working on for the last months, this project was a bit lower on my to-do list. But i'm back for now to give some good news.

For the last few weeks there is a lot of progress made for the new upcoming version: 1.2! This will feature more fixes to problems users have with the application, as well as new minor/major features being added to the project.

One major feature that will be added is suggested by a lot of people: the option to use custom money trees!! And honestly, a few months back I didn't know how to start and what to use to bring this to this app. But nevertheless I started with this a little while back and changed almost all lines of code that refer to the money tree, ran tests on each change, correct the errors I made, test again, make new textures and.. Voilà! Now we are in the last stage of this task and currently it works really well (but still with some bugs).

And also, themes are added to the project. So you can choose between more textures and graphics. Of course, the available options are not unlimited, but maybe there will be more textures being added in future releases.

For those who's interested, here are some screenshots of the new developed money tree + the different themes.

![Schermafbeelding 2024-09-11 203746](https://github.com/user-attachments/assets/db6c416e-2586-4394-9285-5b67dd006da7)
![Schermafbeelding 2024-09-11 202558](https://github.com/user-attachments/assets/f623a45b-214e-4fc0-909b-3767f4b3a2f4)
![Schermafbeelding 2024-09-11 202825](https://github.com/user-attachments/assets/d348cee7-1284-4a57-bb6f-6b9fe7e246f5)

You can check the full package when the new release is out. The date when it will be published will be coming soon.

**Comment by Smoozle** [comment]
Absolutely awesome! Can't wait! Thank you for your hard work!

**Reply by Macronair** [reply]
Thanks!

**Reply by Smoozle** [reply]
Just had an extra thought. The 'Ask the Audience' lifeline, is it possible to set a toggle to _not_ auto display the percentages, as I have a plan to use this via Twitch where we will actually be asking the audience via a live poll and don't want the game to display the correct answer via the percentages?

**Reply by CommunityBaldi** [reply]
> Thanks!

What date did the v1.2 is released?

**Comment by ImmanuelJeswyn** [comment]
This is amazing. Looking forward the new version. Please add ability to change background music.

**Reply by Macronair** [reply]
Thanks! By the way, changing the background music and other sound cues is already possible since the 1.1 release.
Supported formats are .mp3 and .flac at the moment.

**Comment by ImmanuelJeswyn** [comment]
Somehow I missed it. Will Check. Thanks for updating

**Comment by JCEurovisionFan** [comment]
Does it handle currency formats? Some of them separate with a comma for the thousands while others separate with a space bar. Also, some of them placed a currency symbol after the number.

**Reply by Macronair** [reply]
Good question. Currently the used separator in the money amounts depends on what is applied in Windows. But I think that it'll be easier for most people to use a separator other than that Windows will automatically set for you.

As for the currency symbol, this is already an independent setting in the app. Only thing to add is the option to place the symbol right or left of the money amount.

**Comment by davemachala** [comment]
Hi, will it be possible to make the currency sign appear to the right of the amount? 
(like 10 000 Kč as shown in the image below) Many currencies use this notation. Thank you for your work😌

![IMG_9705](https://github.com/user-attachments/assets/488168ff-7bbd-49d8-a1b1-b4b2c0fc607a)
![IMG_9707](https://github.com/user-attachments/assets/0ebad265-5773-449a-a95b-754d5e301246)

**Reply by Macronair** [reply]
Ah yes, it was on my to-do list already. When 1.2 is completed, this option is available!
But thanks for noticing! ;)

**Comment by Macronair** [comment]
Hello all,
Release 1.2 is on it's way and almost complete. But because of the new features and major changes in the code, it is needed to test the program a bit longer. So, unfortunately I need to move the release date to begin November instead of the end of October.

I'm sorry for that, but I also want that 1.2 will be a lot more stable with less errors and bugs so that everyone can use it without any (major) issues.

Of course I will keep you guys updated about the developement process. ;)

**Reply by Smoozle** [reply]
No worries dude! I think everyone here super appreciates what you're doing for us! :D

**Reply by Ralohun** [reply]
Cant wait!!

**Comment by Macronair** [comment]
Developement update:
For the experts in SQL, there is also an option to connect to a SQL server that runs on another computer.
The reason to use it is different per user. Maybe to save some resources on the main PC, or to use the application on different machines for example.

Also for the 1.2 release. SQL Server Express is required to run the software. For remote databases as well as local databases.

Here a sneak peak of the login window when starting the application:
![image](https://github.com/user-attachments/assets/c53fd2a2-dd89-4d2a-ade8-69c05278633e)
